tests/kernel/threads/thread_apis/src/main.c

```diff
@@ -93,6 +97,9 @@ void test_customdata_get_set_coop(void)
 
 	k_sleep(500);
 
+	int ret = strcmp(tid->name, "customdata");
```

Comment:
Would we need a `k_thread_name_get()`?

AFAICS, changes LGTM, could not find anything amiss.

Reply:
thread->name will work from supervisor mode, but if we need to get these names from user mode then this API will need to be added and declared as a system call.